### DISCUSSION TITLE
Custom method without source parameter, only context

### DISCUSSION THREAD

**Discussion by olivierboudet** [mapstruct/mapstruct#2747] [Q&A]
Hello,

I have a mapper using a custom mapping method : 
```
@Mapping(target = "site", source = "sevrage")
abstract SiteDTO toDto(Sevrage sevrage, @Context MappingContext MappingContext);

protected String site(Sevrage sevrage, @Context MappingContext MappingContext) {
  return mappingContext.getSite();
}

```
Which works great. My question is just about the source parameter : is it possible to omit it ? As it is not used in the `site(Sevrage sevrage, @Context MappingContext MappingContext)` method, I would like to remove it, like `protected String site(@Context MappingContext MappingContext)
`
But it does not work, is there any way to do this ? (And avoid to have issues reported by IDE & Sonarqube because of unused parameters)

**Comment by Zegveld** [comment]
Best thing I can think of is using the following for the mapping:
```
@Mapping(target = "site", expression = "java(mappingContext.getSite())")
abstract SiteDTO toDto(Sevrage sevrage, @Context MappingContext mappingContext);
```

**Reply by olivierboudet** [reply]
Thanks, but I am not a fan of using the `expression` attribute. It hides usage of methods when using IDE's features (find usages, refator, etc).

**Reply by Zegveld** [reply]
true, was just wondering why is it actually a `@Context` annotated field? Are there other properties present that you do not want to map in that MappingContext class?
Because then you could just do `abstract SiteDTO toDto(Sevrage sevrage, MappingContext MappingContext);`.
Or perhaps you could introduce an interface like:
```
interface SiteProvider {
    String getSite();
}
```
and then have the method like `abstract SiteDTO toDto(Sevrage sevrage, SiteProvider siteProvider);`
